modules/openapi-generator/src/main/resources/typescript-fetch/apis.mustache

```diff
@@ -306,7 +306,11 @@ export class {{classname}} extends runtime.BaseAPI {
         return new runtime.JSONApiResponse<any>(response);
         {{/isArray}}
         {{#returnSimpleType}}
-        return new runtime.TextApiResponse(response) as any;
+        if (response.headers.get('content-type') === 'application/json') {
```

Comment:
can you introduce a helper function, similar to https://github.com/OpenAPITools/openapi-generator/blob/43493a94fef55e874391760053066c1cc5efd6bb/modules/openapi-generator/src/main/resources/typescript-angular/configuration.mustache#L179?

Reply:
Sure. I will copy the logic you linked.

Reply:
I used the regex you linked but changed the capturing groups to non-capturing groups.